### DISCUSSION TITLE
404 after upgrading from 3.2.1 to 3.3.0

### DISCUSSION THREAD

**Discussion by jemoreno** [bullhorn/career-portal#420] [Q&A] [answered]
INSERT SHORT DESCRIPTION EXPLAINING THE HIGH-LEVEL REASON FOR THE NEW ISSUE HERE.

## Current behavior
After upgrading (removing old plugin and installing new one) when navigating to the /wp-content/plugins/bullhorn-oscp page I get a 404.
-

## Expected behavior
Page should render as expected
-

## Steps to replicate behavior

1. navigate to https://spotontalent.com/wp-content/plugins/bullhorn-oscp/

## Screenshots

__Current__

__Expected__

**Comment by charlesabarnes** [comment]
Hello, to ensure everything is installed correctly, can you view the plugin under the Wordpress admin page?

**Comment by jemoreno** [comment]
Yes, and I've verified the app.json is being generated as well

**Comment by charlesabarnes** [comment]
Yeah I guess I can confirm I do see the json file under `https://spotontalent.com/wp-content/plugins/bullhorn-oscp/app.json`  

What happens if you use the shortcode on a page so `[oscp]` on a random WordPress page? Also, can you confirm that you see an index.html file in that directory?

**Comment by jemoreno** [comment]
yes, the index.html file is present and the shortcode returns the 404 as well, you can click the search jobs button on the original page to see how it also returns a 404

**Comment by jemoreno** [comment]
<img width="599" alt="Screen Shot 2021-09-16 at 12 54 57 PM" src="https://user-images.githubusercontent.com/4119838/133661327-c90789f9-ec3d-4fbc-ba1c-156e46bd90af.png">

**Comment by charlesabarnes** [comment]
I am able to see every other file resolve (minus some security rejections).   I am wondering why the index.html file would not.  My initial instinct is server config in some form.   Was there a .htaccess in this directory previously that helped resolve requests?

**Comment by jemoreno** [comment]
not to my knowledge

**Comment by jemoreno** [comment]
it should have been a fresh install based on the fact that I removed the plugin and then installed it fresh, I didn't see any other configuration steps in the documentation.

**Comment by charlesabarnes** [comment]
Did you update WordPress Versions as well by chance?  I do not see any reason this should be happening.  I am unable to reproduce this specific issue on my [demo site](http://charlesabarnes.com/bullhornWordpress/wp-content/plugins/bullhorn-oscp/#/ ) (error intentional).

The only change that can cause this to my knowledge is a change to your `.htaccess`'s URL rewrite rules or the file permissions for that specific file being off (looks good to me in that screenshot).  

Is getting your web host looped in an option? Because from everything you have shown, it should be working

**Comment by jemoreno** [comment]
We did not upgrade the wordpress version, I was just brought in as a consultant today on this issue so I'm unfamiliar with the host situation yet.

**Comment by jemoreno** [comment, marked as answer]
I tweaked the .htaccess file to get the page to render but every job posting now links to undefined `https://spotontalent.com/wp-content/plugins/bullhorn-oscp/#/jobs/undefined`

**Comment by jemoreno** [comment]
<img width="708" alt="Screen Shot 2021-09-16 at 2 44 28 PM" src="https://user-images.githubusercontent.com/4119838/133676065-7673f8c2-381b-4fb8-a23d-10129d679782.png">

**Comment by jemoreno** [comment]
<img width="1311" alt="Screen Shot 2021-09-16 at 3 19 32 PM" src="https://user-images.githubusercontent.com/4119838/133680292-2edc410d-8728-42d0-a123-3f13cebcd849.png">
This is causing invalid job id

**Comment by charlesabarnes** [comment]
Heya! `id` needs to be added to the fields property in the `app.json` file.  That controls the fields pulled down in the API requests,

**Comment by jemoreno** [comment]
Thanks @charlesabarnes We're up and running again!
